File: Visitor/Ruby/CreditCard.rb
require_relative "IPayment"
require_relative "CPU"
require_relative "VGA"

class CreditCard < IPayment
    def visit(component)
        if component.is_a?(CPU)
            puts "Purchase CPU with Credit Card"
        elsif component.is_a?(VGA)
            puts "Purchase VGA with Credit Card"
        end
    end
end

File: Visitor/Dart/CreditCard.dart
import 'CPU.dart';
import 'IPayment.dart';
import 'VGA.dart';

class CreditCard implements IPayment {
  @override
  void visit(component) {
    if (component is CPU) {
      print("Purchase CPU with Credit Card");
    } else if (component is VGA) {
      print("Purchase VGA with Credit Card");
    }
  }
}
